### PR TITLE
Enhance datasource framework.

### PR DESCRIPTION
**Pull request by louyuting**
### Describe what this PR does / why we need it
Enhance datasource framework.

### Special notes for reviews

### CODE REVIEW

File: ext/datasource/datasource.go [hunk 1]
```diff
@@ -12,9 +12,12 @@ type DataSource interface {
...
-	ReadSource() []byte
-	// Initialize, init datasource and load initial rules
-	// start listener
+	// return source bytes if succeed to read, if not, return error when reading
+	ReadSource() ([]byte, error)
+	// Initialize the datasource and load initial rules
+	// start listener to listen on dynamic source
+	// panic if initialize failed;
+	// once initialized, listener should recover all panic and error.
 	Initialize()
```
Comment (sczyh30) [inline ext/datasource/datasource.go]: Do we need to return the error instead of just panicking?

File: ext/datasource/property.go [hunk 2]
```diff
@@ -51,15 +55,18 @@ func (h *DefaultPropertyHandler) Handle(src []byte) error {
 		}
 	}()
 	// converter to target property
-	realProperty := h.converter(src)
+	realProperty, err := h.converter(src)
+	if err!=nil {
```
Comment (sczyh30) [inline ext/datasource/property.go]: Plz format the code :)

File: ext/datasource/property.go [hunk 3]
```diff
@@ -51,15 +55,18 @@ func (h *DefaultPropertyHandler) Handle(src []byte) error {
 		}
 	}()
 	// converter to target property
```
Comment (sczyh30) [inline ext/datasource/property.go]: Convert to?

File: ext/datasource/datasource.go (hunk 1 above)
Comment (louyuting) [reply ext/datasource/datasource.go]: Initialize function should be executed automatically when new a DataSource and the user doesn't need to call it manually.

File: ext/datasource/property.go (hunk 3 above)
Comment (louyuting) [reply ext/datasource/property.go]: Yes, I will polish here.

File: ext/datasource/property.go (hunk 2 above)
Comment (louyuting) [reply ext/datasource/property.go]: ok~

**Comment by sczyh30** [comment]
Thanks!

**Pull request merged**